### PR TITLE
add sorted list + binary search insert/delete version of qkchash

### PR DESCRIPTION
**Pull request by qizhou**
Using a sorted vector to find ith smallest element
Using binary search to insert an element to keep the vector's elements ordered

Performance: (for 64K entries, 1000 hashes)
DST: 10s
Sorted vector: 20s

The sorted vector code will be used in python (for verification purpose)

### CODE REVIEW

File: qkchash/qkchash.cpp [hunk 1]
```diff
@@ -94,11 +95,49 @@ void qkc_hash(
...
+        uint64_t p = fnv64(i ^ seed[0], mix[i % mix.size()]);
+        for (uint32_t j = 0; j < mix.size(); j++) {
+            // Find the pth element and remove it
+            uint32_t idx = p % slist.size();
+            new_data[j] = slist[idx];
+            slist.erase(slist.begin() + idx);
+
+            // Generate random data and insert it
+            p = fnv64(p, new_data[j]);
+            slist.insert(std::upper_bound(slist.begin(), slist.end(), p), p);
```
Comment (qcgg) [inline qkchash/qkchash.cpp]: make sure we don't add duplicated element.

Comment (qizhou) [reply qkchash/qkchash.cpp]: Thanks for pointing this out.   Fixed in the new code.  Please take a look.

Comment (qcdll) [reply qkchash/qkchash.cpp]: I don't see the downside of keeping duplicates (except it would be different from the ordered set implementation)

Comment (qizhou) [reply qkchash/qkchash.cpp]: The downside is that for ordered set implementation, we have to keep the reference count of a key if a key is inserted twice or more.  In addition, implementing an ordered set with duplicated keys is more complicated.

**Pull request merged**